dali/kernels/signal/decibel/decibel_calculator.h

```diff
@@ -25,29 +25,36 @@ namespace signal {
 template <typename T>
 struct DecibelCalculator {
  public:
+  DALI_HOST_DEV DALI_FORCEINLINE
   explicit DecibelCalculator(T mul = 10.0, T s_ref = 1.0, T min_ratio = 1e-8)
-      : inv_mul_(1. / mul)
+      : inv_mul_(T(1) / mul)
       , mul_log2_(mul * kLog2Factor)
       , s_ref_(s_ref)
-      , inv_s_ref_(s_ref == 1.0 ? 1.0 : 1.0 / s_ref)
+      , inv_s_ref_(s_ref == T(1) ? T(1) : T(1) / s_ref)
       , min_ratio_(min_ratio) {
-    assert(min_ratio_ > 0.0);
-    assert(inv_s_ref_ > 0.0);
+    assert(min_ratio_ > T(0));
```

Comment:
Make sure that this assert doesn't restrict your launch paramters (block size) in Debug build - try to run the kernel with 1024-thread block. If it fails in debug but runs in release, remove asserts or put them in `#ifndef __CUDA_ARCH__`. I've recently stumbled upon the issue of asserts restricting launch parameters.

Reply:
Checked, there is no problem